### DISCUSSION TITLE
Some Buttons activate not one, but two LED's

### DISCUSSION THREAD

**Discussion by TAJDW** [shanteacontrols/OpenDeck#253] [General]
Hey paradajz,

Great Platform!
Currently woking with M board v2.2.1, firmware v7.6.0, and I'm experiencing some odd behavior with button- and LED configuration. I have this problem where some of the buttons activate 2 LEDs, and some LEDs that don't respond to "LED color testing" in the OpenDeck Configurator.

To be precise; the 'LEDs that are expected to activate exclusively' are indexed at: 4, 6, 8, 12, 14, 16. 
And the non-responsive leds are indexed at: 5, 7, 9, 13, 15, 17. 

I made sure it is not on the hardware side of things, as i attached multiple LED wiring-setups to the the board, and the issue persists. And I know for sure the LEDs are not broken. Therefor i began to think that the issue might as well come from the sourcecode somewhere..?

I could send you more detailed information like my .sysex configurator file and hardware setup etc., but maybe this is a known issue in the firmware?

Hope you can help with this!

Cheers, 
Tim

**Comment by paradajz** [comment]
What kind of LEDs are you using? Also, what kind of buttons? Can you post a picture of your setup?

**Comment by paradajz** [comment]
Also I am not sure I undestand how to reproduce this. I grabbed my testing M board with the latest firmware, ran a factory reset on it, connected LED 5 (R1, C6), enabled color testing for it in the configurator and it turned on. Same for LED 7 etc.

**Comment by TAJDW** [comment]
I am using single color red LEDs. And the buttons i am using with that are: Sparkfun's [Button Pad 2x2 Breakoutboard](https://www.sparkfun.com/products/9277) and silicon conductive button pad [silicon conductive button pad](https://www.sparkfun.com/products/7836). 

Setup Top:
![Setup_TOP](https://github.com/shanteacontrols/OpenDeck/assets/153218868/95a595cf-d744-4607-92b2-1f1f2fc637e4)


Setup Bottom:
![Setup_Bottom](https://github.com/shanteacontrols/OpenDeck/assets/153218868/5727c4c9-b506-4496-a66b-fbcd5c9872d6)


sketchy LED schematic (top view):
![LED Schematic](https://github.com/shanteacontrols/OpenDeck/assets/153218868/9d7dc9a6-a5f7-4361-b837-44df85440091)

Hope this helps!

**Reply by TAJDW** [reply]
And don't know if this will be usable, but here i captured my LED/Button mapping setup with the Configurator:
[LEDButton Mapping.pdf](https://github.com/shanteacontrols/OpenDeck/files/13624050/LEDButton.Mapping.pdf)

**Comment by paradajz** [comment]
From the schematic of that breakout, it appears the LEDs use common anode. So what this means is that you need to keep all the LEDs on a single breakout connected to the same row on the OpenDeck board. Also each LED should then be connected to a dedicated column pin. Did you try that?

**Reply by TAJDW** [reply]
I tried using the breakout in an unconventional way, so that they would share common cathode instead. Just for ease of use.. Since the other LEDbuttons work fine I couldn't figure it out.
So what you suggest is matching the column-rows to the right breakout board, or just rewire the so that each LED is connected to the right pins on the OpenDeck bord? I'm gonna give that a try!

And what i cant seem to figure out is whether the SEG_A to SEG_G pins on the MAX2719 IC are the same as groundpins, or that they differ and shouldn't be interconnected

Thanks for the response!

**Reply by paradajz** [reply]
> that they would share common cathode instead

That will not work because all their anodes are connected together already.

There is no ground concept on either the digital outputs or digital inputs on the OpenDeck board. Both use matrix multiplexing.

**Reply by TAJDW** [reply]
Oke I see, That explains it. I'm gonna try what you previously stated and see what comes from that!
Thanks for the quick reply!

**Comment by TAJDW** [comment]
Hey paradajz,

Turned out that to be my wiring after all! So im going to rewire a bit and then enjoy my project with your amazing platform! Due to my lack of knowledge concerning multiplexers I couldn't figure this out by myself.. So

Thank you so much for your help!

**Reply by paradajz** [reply]
Glad it's resolved.
